### DISCUSSION TITLE
Running inference without datasets

### DISCUSSION THREAD

**Discussion by jama1017** [hkchengrex/XMem#63] [Q&A]
Hi,

Thanks for putting out this amazing work. I wonder if it is possible to directly use the pretrained model to run inferences on my own videos without going through the GUI or use `eval.py`, which requires me to download all the datasets? I guess my question is more about how I can do that if possible.

**Comment by hkchengrex** [comment]
`eval.py` does not require users to download ~~all~~ any dataset. You can structure your data as https://github.com/hkchengrex/XMem/blob/main/docs/INFERENCE.md#on-custom-data

And run `python eval.py --output [your_output_directory] --dataset G --generic_path [your_data_root] --mem_every 10` (or with mem_every 5)

**Reply by jama1017** [reply]
Thanks so much for the clarification!

**Reply by jama1017** [reply]
Just as a follow up question - for the custom dataset, do we need to add annotation images for each of the video frames? Can we add no annotations at all? I tried the command you gave and it only processed the frames for which the annotations are provided.

**Reply by hkchengrex** [reply]
It should process all the frames since `save_all` should be true when dataset is `G`. I guess you can try passing `--save_all` but what you said is not expected. One annotation per video should be enough.

**Reply by jama1017** [reply]
Thanks for the quick reply. I figured out why - I didn't put in the annotation image for the first frame of the video but put in the annotation image for the last frame instead, so it only processed that last frame.
